src/hmem_cuda.c

```diff
@@ -102,6 +102,14 @@ CUresult ofi_cuInit(unsigned int flags)
 
 int cuda_copy_to_dev(uint64_t device, void *dev, const void *host, size_t size)
 {
+#ifdef HAVE_GDRCOPY
+	int err;
+
+	err = cuda_gdrcopy_to_dev(device, dev, host, size);
+	if (!err)
+		return 0; 
+#endif
```

Comment:
ifdefs in the middle of functions make it a pain to read the code.  It's preferable to always define cuda_gdrcopy_to_dev, and just make it a no-op or always fail if gdrcopy is not enabled.  The compiler should toss out the resulting code.

Separate, why does this fall through to cuda memcpy on error?  The commit message does not describe why we want that, since gdrcopy is listed as an alternative, not in addition to.

Reply:
Removed the ifdefs in the code.

In the case gdrcopy is not available, we still want to use cudaMemcpy.